pyhive/tests/test_hive.py

```diff
@@ -72,6 +72,21 @@ def test_complex(self, cursor):
             '0.1',
         ]])
 
+    @with_cursor
+    def test_async(self, cursor):
+        cursor.execute('SELECT * FROM one_row', async=True)
+        while cursor.poll() != ttypes.TOperationState.FINISHED_STATE:
+            cursor.fetch_logs()
+        
+        self.assertEqual(len(cursor.fetchall()), 1)
+    
+    @with_cursor
+    def test_cancel(self, cursor):
+        cursor.execute('SELECT * FROM one_row', async=True)
+        cursor.cancel()
```

Comment:
It looks like this test fails
AssertionError: 2 != 3

finished != canceled

assuming this is just a race condition, I propose we cheat by making Hive call Thread.sleep
reflect("java.lang.Thread", "sleep", 10 * 1000)